### PR TITLE
[MIRROR] [Ready] Punching/Grabbing Rebalance

### PR DESCRIPTION
**Pull request by ghost**
Original PR: https://github.com/tgstation/tgstation/pull/43291
----------------------------
:cl:
balance: Humans now have a punch damage range of 1-10, up from 0-9
balance: The chance to miss a punch now increases the more stamina and brute damage you have
balance: The stun duration for a critical punch now scales with how much stamina/brute damage the victim has
balance: Punches now deal full stamina damage + half damage in brute (effectively 1.5x damage on all punches, not accounting for stamina regeneration)
balance: Kicking (punching a prone person) does full brute damage, never misses, and does 1.5x damage
balance: Grabs are now harder to break out of the more stamina/brute damage you have
balance: Failing to resist out of a grab will deal a bit of stamina damage to you
balance: In all instances above where both stamina and brute damage are considered, brute is weighed 1/2 as heavily as stamina (meaning stamina has more of an effect than brute when it comes to punch-stun duration, miss chance, etc)
/:cl:

Please see https://tgstation13.org/phpBB/viewtopic.php?f=9&t=22154#p485654 for the initial discussion thread

__Grab change goals:__
- Make choking worth going for without making it too useful in any scenario
- Make restraining weakened people with grabs more reliable for things like kicking out assistants non-lethally

__Punch change goals:__
- Make fistfights less "I got an RNG stunpunch, I win!"
- Give fistfights a nonlethal way of ending instead of one person inevitably going into crit (stamina stunning, stamina damage punches -> hard-to-break-free-of grab)
- Make punching play into the new grab changes by dealing stamina damage (see above point)
- Retain the lethality of fistfights should the participants deliberately take it to the death (1.5x damage on everything, potential for longer stuns, more raw damage when kicking)

### CODE REVIEW

**Comment by k-k-karma** [comment]
>remove RNG from disarms
>add it to punches
tg

**Comment by Majick1** [comment]
+1

**Comment by mikolaj007** [comment]
+1

**Comment by Manly-Dwarf** [comment]
I would rather punches be consistent damage, honestly. It's not a good thing when your punch has a random chance to deal a fuckload of damage, because running up at a guy and just punching him a couple times is a reliable way to annoy people without critting them.
The miss chance is dumb, I fail to see why it needed a change. It's not like we're goon where you can sprint.

> Make fistfights less "I got an RNG stunpunch, I win!"
>proceeds to buff stunpunch
>tg

Kick change also buffs the stunpunch. Why?
Grab change is nice, it causing stamina damage is not so nice. We should revert back to our old code there, honestly, been meaning to do that since pull/grab being merged was a mistake.
I don't have an opinion on the final stamina change since I don't have experience with this PR, but it reminds me of the time Calibraptor or whoever did something similar on oldbase, so it could be quite nice. I'd remain optimistic about that.

All in all it might be worth giving this PR a shot. The things I don't like about it probably won't matter much in the grand scheme of things.

Can you designers say somethin more than +1?

**Comment by mikolaj007** [comment]
Alright, ill detract my +1 and actually think long and hard on it:
I like the kicks, but they could use less randomness and i feel like they should be more of a way to keep a person down rather than just deal 1.5-15 damage.
Im impartial on the hit chance, but i think it just fucks up the loser even harder for no reason other than that hes already losing. I approve of grab changes because literally nobody goes for a grab other than to table someone and keep them stunlocked. I say tabling should get a bit nerfed, because being able to perma-stunlock someone with a table while funny isnt good game design. Back to my point grabbing seems like an ok-ish alternative to subdue someone if you have no cuffs, and get the stun crit. 
+0 for now
+1 if we can nerf either rng stuns or kicking, but id prefer to nerf rng stuns

**Comment by mikolaj007** [comment]
+1
after a few more tests and thinking about it, this decreases the lethality of fistfights considerably

> I would rather punches be consistent damage, honestly. It's not a good thing when your punch has a random chance to deal a fuckload of damage, because running up at a guy and just punching him a couple times is a reliable way to annoy people without critting them.

while i absolutely agree fists should deal consistent damage, it still deals between 0.5-5 brute damage on a hit, so thats not near a crit as you think it is

> The miss chance is dumb, I fail to see why it needed a change. It's not like we're goon where you can sprint.

ye, i agree miss chance is pointless RNG, its not that bad, and even then we can modularize it out

> > Make fistfights less "I got an RNG stunpunch, I win!"
> > proceeds to buff stunpunch
> > tg
> 
> Kick change also buffs the stunpunch. Why?

kick makes sense from a realism perspective, of course when a person is knocked down, they are gonna get their shit kicked in. even then, its still 1.5-15 brute damage, so its not enough to actually kill anyone in a knockdown or two.

> Grab change is nice, it causing stamina damage is not so nice. We should revert back to our old code there, honestly, been meaning to do that since pull/grab being merged was a mistake.

i sadly cant really test this alone

> I don't have an opinion on the final stamina change since I don't have experience with this PR, but it reminds me of the time Calibraptor or whoever did something similar on oldbase, so it could be quite nice. I'd remain optimistic about that.

i actually like the solution of using stamina damage
you deal 1.5-15 damage, but only a third of it is brute
so, basically once a person is horizontal, you have to keep hitting them 2x more than you already have to actually crit them

**Comment by ghost** [comment]
fuck why is this not modularised yet

**Comment by ghost** [comment]
@mikolaj007 nice review, some great insight into this PR

**Comment by bigbluebeaker** [comment]
+1 give her the ol test run

**Comment by ExpAntique** [comment]
I like the idea of getting weaker the lower your health/stamina is, but the rest is pretty gay.

**Comment by mikolaj007** [comment]
can you elaborate on the "pretty gay" parts?

**Review by yoyobatty — APPROVED** [review]
